### ISSUE TITLE
Channel Announcement features

### ISSUE THREAD

**Issue by t-bast** [lightningnetwork/lightning-rfc#769]
The [channel_announcement](https://github.com/lightningnetwork/lightning-rfc/blob/master/07-routing-gossip.md#the-channel_announcement-message) message contains a `features` field, which is currently unused.

The requirements state that:

```
The origin node:
- MUST set features based on what features were negotiated for this channel, according to BOLT #9
```

However `what features were negotiated for this channel` is very vague, since none of the channel creation messages (`open_channel`, `accept_channel`) contain a `features` field. Choosing whether a feature should be set as `optional` or `mandatory` in `channel_announcement` feels a bit weird, because the features we currently have apply to the channel's whole lifetime.

Why is this important? Because when you receive an [announcement_signatures](https://github.com/lightningnetwork/lightning-rfc/blob/master/07-routing-gossip.md#the-announcement_signatures-message) message, you need to generate the same `channel_announcement` as your peer to validate the signatures, so you need to figure out what bits your peer will have put in `features`.

There are currently two features that may have their place in the `channel_announcement`:

- `option_static_remotekey`
- `option_support_large_channel`

Bolt 9 specifies that `option_support_large_channel` is `INC+`: that means every wumbo channels should have bit 18 set. Right now eclair doesn't set this bit. Does lnd and c-lightning do?

Why isn't `option_static_remotekey` flagged as a `channel_announcement` feature in Bolt 9 too?

I think we should clarify `channel_announcement.features` now, WDYT @rustyrussell @Roasbeef @cfromknecht @TheBlueMatt @ariard?

**Comment by TheBlueMatt** [comment]
`channel_announcement` features are all about what types of payments others can send through that channel. `static_remoteky` doesn't change that at all so isn't relevant. Why `option_support_large_channel` is "always required" I have no idea - that seems to just be a typo - it should be "always optional".

**Comment by TheBlueMatt** [comment]
Opened a PR to fix `option_support_large_channel` - #770.

**Comment by t-bast** [comment]
I don't understand `option_large_channel required` to mean that only large payments will be routed through that channel, `htlc_min` is here to provide that kind of information.

I was interpreting `channel_announcement.features` more as "structural information about that channel", which is why I thought `option_static_remotekey` could make sense. But it's true that remote nodes don't care about that information. But they don't care about `option_support_large_channel` either, they just want to know the channel's capacity, so why even put `option_support_large_channel` in the `channel_announcement`?

**Comment by TheBlueMatt** [comment]
> But they don't care about option_support_large_channel either, they just want to know the channel's capacity, so why even put option_support_large_channel in the channel_announcement?

Because the capacity is not available in the channel_announcement? You only get the capacity if you scan the chain to lookup the UTXO associated with the channel, which not all nodes have. More importantly, maybe, it just feels...icky to decide to violate the "> 2**24" spec on the basis of a UTXO lookup.

> So far the only upcoming feature I could see added to channel_announcement.features if that field is only used for payment-relay-related stuff would be option_payment_points (or option_ptlc, whatever we call the PTLC-based payments)

Aside from the above, I think I generally agree.

**Comment by t-bast** [comment]
> Because the capacity is not available in the channel_announcement? You only get the capacity if you scan the chain to lookup the UTXO associated with the channel, which not all nodes have. More importantly, maybe, it just feels...icky to decide to violate the "> 2**24" spec on the basis of a UTXO lookup.

Fair enough, but even if you don't look up the UTXO to get the capacity, when you want to send payments you should only be looking at `htlc_maximum_msat` then? Why do you care that it's a wumbo channel? Or are you using the `support_large_channel` indication to potentially send multiple big HTLCs (on the order of `htlc_maximum_msat`) through that channel for an MPP payment (that wouldn't be too crazy)?

**Comment by cdecker** [comment]
I agree with @t-bast, we already have all the signaling in place as part of the `channel_announcement` and `channel_update` to signal what the acceptable size of an HTLC is, and we can add more fields to that to indicate the maximum concurrency etc.

The `option_large_channel` flag is only useful during channel creation, and should not be reflected in the `channel_announcement`.

**Comment by cdecker** [comment]
Addressed by #773
